### PR TITLE
nbibtex binary to open the 'os' package

### PR DESCRIPTION
**Pull request by nrnrnr**
This fix solves the immediate problem that after the `nbibtex` binary is built, it fails on startup.

### CODE REVIEW

**Comment by nrnrnr** [comment]
I ran a quick test on an existing document.  Output looks good.

This project needs a regression test :-(

**Comment by barak** [comment]
Thanks! Will push into Debian as soon as I log it...

Merged, but on a different branch ("upstream") that is missing the debian/ subdir.

**Comment by barak** [comment]
PS bugs.debian.org/nbibtex shows a need by bib2html for nrlib.lua

```
$ bib2html
lua5.1: ./bibtex.lua:20: module 'nrlib' not found:
    no field package.preload['nrlib']
    no file './nrlib.lua'
    no file '/usr/local/share/lua/5.1/nrlib.lua'
    no file '/usr/local/share/lua/5.1/nrlib/init.lua'
    no file '/usr/local/lib/lua/5.1/nrlib.lua'
    no file '/usr/local/lib/lua/5.1/nrlib/init.lua'
    no file '/usr/share/lua/5.1/nrlib.lua'
    no file '/usr/share/lua/5.1/nrlib/init.lua'
    no file './nrlib.so'
    no file '/usr/local/lib/lua/5.1/nrlib.so'
    no file '/usr/lib/x86_64-linux-gnu/lua/5.1/nrlib.so'
    no file '/usr/lib/lua/5.1/nrlib.so'
    no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
    [C]: in function 'require'
    ./bibtex.lua:20: in main chunk
    [C]: in function 'require'
    /usr/bin/bib2html:6: in main chunk
    [C]: ?

$ apt-file search nrlib.lua
$ echo nada
```

**Comment by nrnrnr** [comment]
This one's pretty dire.  bib2html needs to load a .so file containing the bibtex module (which is statically linked into the nbibtex binary).  My machine shows a .so last built in 2006.

Where would be the right place to install a .so file?

**Comment by barak** [comment]
The natural place would be `/usr/lib/x86_64-linux-gnu/lua/5.1/nrlib.so` or the equivalent on other architectures. In autotools terminology, it would be in `${pkglibexecdir}/`. Which is on the dynamic library search path above. (As are some other security-scary places.)

**Comment by nrnrnr** [comment]
This seems a little odd to me, as what's to prevent name conflicts with .so files from other Debian packages?  As contrasted with `/usr/share/bibtex` where the architecture-independent files live.  

`bib2html` is going to load this file using its absolute pathname, so there's no need to put it in a directory searched by other tools.  But if you say this is where to put it, this is where it shall go!

**Comment by barak** [comment]
Agreed. The directory part of `/usr/lib/x86_64-linux-gnu/lua/5.1/nrlib.so` is package-specific, but for the lua 5.1 package. The `${pkglibexecdir}` here would be `/usr/lib/${ARCH}/nbibtex/`, which would be the right place. Unless the library is of more general utility, in which case it should be split off into its own package I suppose.

**Comment by nrnrnr** [comment]
The particular `nrlib` is a thing that supports Lua 5.0 compatibility.  I'm going to purge it from the package.

I think my current plan is `${prefix}/lib/${ARCH}/lua/5.1/nbibtex/mumble.so`, which should cover all the bases.

Is there a Debian standard way to discover `${ARCH}` in my `configure` script?

**Comment by barak** [comment]
> Is there a Debian standard way to discover `${ARCH}` in my `configure` script?

Because you might be in a cross-compilation situation, you want the "host" architecture, rather than the "build" or "target" architectures. (There are three because you might be compiling a compiler on BUILD which will run on HOST and generate executables for TARGET, and all three of these might have different architectures.)

You could get it from

```
$ dpkg-architecture -qDEB_HOST_GNU_TYPE
x86_64-linux-gnu
```

During the build, the base directory you want is passed in option `--libdir`

```
$ debian/rules build
dh build --parallel
   dh_testdir -O--parallel
   dh_auto_configure -O--parallel
warning: Unknown option --build=x86_64-linux-gnu
warning: Unknown option --includedir=${prefix}/include
warning: Unknown option --infodir=${prefix}/share/info
warning: Unknown option --sysconfdir=/etc
warning: Unknown option --localstatedir=/var
warning: Unknown option --libdir=${prefix}/lib/x86_64-linux-gnu
warning: Unknown option --libexecdir=${prefix}/lib/x86_64-linux-gnu
warning: Unknown option --disable-maintainer-mode
warning: Unknown option --disable-dependency-tracking
```

so it would make sense to parse that out and let it take precedence.

If you want something that will work on non-Debian-derived systems, you can use the `autotools` mechanism yourself,

```
$ /usr/share/misc/config.guess | sed s:-unknown-:-:
x86_64-linux-gnu
```

even though, technically speaking, this gives you the "build" architecture rather than the "host" architecture.

You could try to piece it together from the output of `uname` but if you try to do this robustly you'll find there's a reason that

```
$ wc -l /usr/share/misc/config.*
 1420 /usr/share/misc/config.guess
 1799 /usr/share/misc/config.sub
 3219 total
```

TLDR: lots of ways all a bit ugly.

**Comment by nrnrnr** [comment]
`dpkg-architecture` looks perfect; thanks!
